src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs

```diff
@@ -273,9 +274,11 @@ private TaskSeriesCommandResult CreateSucceededResult()
             }
         }
 
-        private void OnMessageAddedToPoisonQueue(object sender, EventArgs e)
+        private void OnMessageAddedToPoisonQueue(object sender, PoisonMessageEventArgs e)
         {
-            _sharedWatcher.Notify(_poisonQueue.Name);
+            // TODO: this is assuming that the poison queue is in the same
+            // storage account
+            _sharedWatcher.Notify(e.PoisonQueue.Name);
```

Comment:
I want to create a bug for this. Not an real issue, but something we should look to improve. This shared watcher notify stuff is just a way to wake up a sleeping queue listener faster when something is written to it on the same instance, so we don't have to wait for the exponential backoff interval.